ddtrace/tracer/option.go

```diff
@@ -851,6 +849,13 @@ func WithAgentAddr(addr string) StartOption {
 	}
 }
 
+// WithAgentTimeout sets the timeout for the agent connection. Timeout is in seconds.
+func WithAgentTimeout(timeout int) StartOption {
+	return func(c *config) {
+		c.httpClientTimeout = time.Duration(timeout) * time.Second
+	}
+}
```

Comment:
Could this option function be unexported?

Reply:
I think it's better for different products, including inside DD, to have it available. I was wary of adding an env variable, so I opted for the tracer option.